sys/linux/socket_netlink_route.txt

```diff
@@ -879,6 +882,114 @@ rtcanmsg {
 
 rtcanmsg_flags = CGW_FLAGS_CAN_ECHO, CGW_FLAGS_CAN_SRC_TSTAMP, CGW_FLAGS_CAN_IIF_TX_OK
 
+lwtunnel_policy [
+	MPLS_IPTUNNEL_DST	nlattr[MPLS_IPTUNNEL_DST, array[mpls_label]]
+	MPLS_IPTUNNEL_TTL	nlattr[MPLS_IPTUNNEL_TTL, int8]
+
+	LWTUNNEL_IP_ID		nlattr[LWTUNNEL_IP_ID, int64be]
+	LWTUNNEL_IP_DST		nlattr[LWTUNNEL_IP_DST, ipv4_addr]
+	LWTUNNEL_IP_SRC		nlattr[LWTUNNEL_IP_SRC, ipv4_addr]
+	LWTUNNEL_IP_TTL		nlattr[LWTUNNEL_IP_TTL, int8]
+	LWTUNNEL_IP_TOS		nlattr[LWTUNNEL_IP_TOS, int8]
+	LWTUNNEL_IP_FLAGS	nlattr[LWTUNNEL_IP_FLAGS, flags[lwtunnel_ip_flags, int16be]]
+	LWTUNNEL_IP_OPTS	nlnest[LWTUNNEL_IP_OPTS, ip_opts_policy]
+
+	ILA_ATTR_LOCATOR	nlattr[ILA_ATTR_LOCATOR, int64]
+	ILA_ATTR_CSUM_MODE	nlnest[ILA_ATTR_CSUM_MODE, ila_attr_csum_mode_policy]
+	ILA_ATTR_IDENT_TYPE	nlnest[ILA_ATTR_IDENT_TYPE, ila_attr_ident_type_policy]
+	ILA_ATTR_HOOK_TYPE	nlnest[ILA_ATTR_HOOK_TYPE, ila_attr_hook_type_policy]
+
+	LWTUNNEL_IP6_ID		nlattr[LWTUNNEL_IP6_ID, int64be]
+	LWTUNNEL_IP6_DST	nlattr[LWTUNNEL_IP6_DST, ipv6_addr]
+	LWTUNNEL_IP6_SRC	nlattr[LWTUNNEL_IP6_SRC, ipv6_addr]
+	LWTUNNEL_IP6_HOPLIMIT	nlattr[LWTUNNEL_IP6_HOPLIMIT, int8]
+	LWTUNNEL_IP6_TC		nlattr[LWTUNNEL_IP6_TC, int8]
+	LWTUNNEL_IP6_FLAGS	nlattr[LWTUNNEL_IP6_FLAGS, flags[lwtunnel_ip_flags, int16be]]
+	LWTUNNEL_IP6_OPTS	nlattr[LWTUNNEL_IP6_OPTS, ip_opts_policy]
+
+# TODO: disable this comment and the header include one, after kernel adds
+# an include for ipv6hdr in the same file
+#	SEG6_IPTUNNEL_SRH	nlattr[SEG6_IPTUNNEL_SRH, ipv6_sr_hdr]
+	LWT_BPF_IN		nlnest[LWT_BPF_IN, bpf_prog_policy]
+	LWT_BPF_OUT		nlnest[LWT_BPF_OUT, bpf_prog_policy]
+	LWT_BPF_XMIT		nlnest[LWT_BPF_XMIT, bpf_prog_policy]
+	LWT_BPF_XMIT_HEADROOM	nlattr[LWT_BPF_XMIT_HEADROOM, int32[0:LWT_BPF_MAX_HEADROOM]]
+
+	SEG6_LOCAL_ACTION	nlattr[SEG6_LOCAL_ACTION, int32[SEG6_LOCAL_ACTION_END:SEG6_LOCAL_ACTION_END_BPF]]
+	SEG6_LOCAL_SRH		nlattr[SEG6_LOCAL_SRH, ipv6_sr_hdr]
+	SEG6_LOCAL_TABLE	nlattr[SEG6_LOCAL_TABLE, int32]
+	SEG6_LOCAL_NH4		nlattr[SEG6_LOCAL_NH4, ipv4_addr]
+	SEG6_LOCAL_NH6		nlattr[SEG6_LOCAL_NH6, ipv6_addr]
+	SEG6_LOCAL_IIF		nlattr[SEG6_LOCAL_IIF, int32]
+	SEG6_LOCAL_OIF		nlattr[SEG6_LOCAL_OIF, int32]
+	SEG6_LOCAL_BPF		nlnest[SEG6_LOCAL_BPF, bpf_prog_policy_seg6_local]
+
+	RPL_IPTUNNEL_SRH	nlattr[RPL_IPTUNNEL_SRH, ipv6_rpl_sr_hdr]
+] [varlen]
+
+lwtunnel_ip_flags = TUNNEL_GENEVE_OPT, TUNNEL_VXLAN_OPT, TUNNEL_ERSPAN_OPT
+
+ip_opts_policy [
+	LWTUNNEL_IP_OPTS_GENEVE	nlnest[LWTUNNEL_IP_OPTS_GENEVE, geneve_opt_policy_lwt]
+	LWTUNNEL_IP_OPTS_VXLAN	nlnest[LWTUNNEL_IP_OPTS_VXLAN, vxlan_opt_policy_lwt]
+	LWTUNNEL_IP_OPTS_ERSPAN	nlnest[LWTUNNEL_IP_OPTS_ERSPAN, erspan_opt_policy_lwt]
+] [varlen]
+
+geneve_opt_policy_lwt [
+	LWTUNNEL_IP_OPT_GENEVE_CLASS	nlattr[LWTUNNEL_IP_OPT_GENEVE_CLASS, int16be]
+	LWTUNNEL_IP_OPT_GENEVE_TYPE	nlattr[LWTUNNEL_IP_OPT_GENEVE_TYPE, int8]
+	LWTUNNEL_IP_OPT_GENEVE_DATA	nlattr[LWTUNNEL_IP_OPT_GENEVE_DATA, array[int8, 128]]
```

Comment:
I don't see any constraints on data length, are there any? If not, use `array[int8]`.

Reply:
I found here that .len is 128
https://elixir.bootlin.com/linux/latest/source/net/ipv4/ip_tunnel_core.c#L238

Reply:
Ah, this is a maximum length AFAIU: https://elixir.bootlin.com/linux/v5.7.8/source/lib/nlattr.c#L258

So `array[int8, 0:128]`.